front/src/applications/rollingStockEditor/components/RollingStockEditorCard.tsx

```diff
@@ -1,57 +1,52 @@
-import React, { SetStateAction, useState, Dispatch } from 'react';
+import React, { useState } from 'react';
 import { RollingStock } from 'common/api/osrdEditoastApi';
 import RollingStockCardDetail, {
   listCurvesComfort,
 } from 'common/RollingStockSelector/RollingStockCardDetail';
 import { RollingStockInfo } from 'common/RollingStockSelector/RollingStockHelpers';
 import RollingStockCurve from 'common/RollingStockSelector/RollingStockCurves';
 import RollingStock2Img from 'common/RollingStockSelector/RollingStock2Img';
-import RollingStockEditorForm from './RollingStockEditorForm';
 
 type RollingStockEditorCardProps = {
+  id: number;
   isEditing: boolean;
-  setIsEditing: Dispatch<SetStateAction<boolean>>;
   rollingStock: RollingStock;
 };
 
 export default function RollingStockEditorCard({
+  id,
```

Comment:
When we take the id of the rollingstock passed in props, we retrieve the one called in **RollingStockEditor**. There is also a call to retrieve the rolling stock in **RollingStockCardDetail**. As we don't immediately have the new id of the `selectedRollingStock` called in **RollingStockEditor**, there's this double call that's made with the old id (deleted rollingstock) + the new id (opened one), which generates an error. To avoid this, I pass the id of the rollingStock that was opened instead.